## Outline

- Null / Alternative Hypothesis
- One sample t-test / z-test
- p-value and Significance Level
- Python for One sample t-test / z-test

**Question 1**

State the null hypothesis and the alternative hypothesis for the following scenarios:

1. A burger place claims the fat content of their bugers is no more than 20%. You have collected a sample of burgers to verify the claim.

   <br>
   
2. With a month to go in the election, the opinion poll shows that the Democrate candidate is leading with 55% support. You want to know if the true percentage is different from 55%.

   <br>
   
3. Apple Inc. is interested to find out if the battery life of a Macbook Pro that has been used for a month is more than 48 hours.

#### 1.1
$H_o: µ \leq 20\% \\ H_a: µ > 20\%$

#### 1.2

$H_o: µ = .55 \\ H_a: µ \neq .55$

#### 1.3

$H_o: µ \leq 48 \\ H_a: µ > 48$


**Question 2**

For each of the scenarios in **Question 1**, decide if you would use a one-sample t-test or a z-test to test your null hypothesis.

#### 1.1

one sample, one tail t-test

#### 1.2

two tailed z-test

#### 1.3

one sample, one tail t-test

**Question 3**

Given a set of real data, you can conduct a **t-test** and **z-test** in Python:

- **t-test** (documentation [here](http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.ttest_1samp.html))
  ```python
  from scipy.stats import ttest_1samp
  t_statistic, two_tailed_p_value = ttest_1samp(sample, null_mean)
  ```

- **z-test** (documentation [here](http://statsmodels.sourceforge.net/devel/generated/statsmodels.stats.proportion.proportions_ztest.html#statsmodels.stats.proportion.proportions_ztest))
  ```python
  from statsmodels.stats.proportion import proportions_ztest
  z_statistic, two_tailed_p_value = proportions_ztest(count_of_success_in_sample, num_of trials, 
                                                       null_proportion)
  ```


Compute the p-values and draw conclusions about hypothesis tests for the scenarios listed in **Question 1**. You are provided the data for each of the scenarios below.

1. https://s3-us-west-2.amazonaws.com/dsci/6002/data/fat_content.csv
2. https://s3-us-west-2.amazonaws.com/dsci/6002/data/election.csv
3. https://s3-us-west-2.amazonaws.com/dsci/6002/data/battery_life.csv

In [1]:
from scipy.stats import ttest_1samp
from statsmodels.stats.proportion import proportions_ztest
import pandas as pd

In [7]:
import scipy as sp

In [12]:
#1. A burger place claims the fat content of their bugers is no more than 20%. 
# You have collected a sample of burgers to verify the claim.
fat = pd.read_csv('https://s3-us-west-2.amazonaws.com/dsci/6002/data/fat_content.csv')
fat_t_stat, fat_two_tailed_p_value = ttest_1samp(fat, 20)
print("t-stat", fat_t_stat)
print("mean:", fat.mean()[0])
print("two-tailed p-value", fat_two_tailed_p_value)
print("one-tailed p-value", fat_two_tailed_p_value/2)

t-stat [ 0.81883086]
mean: 20.4177173378
two-tailed p-value [ 0.41785849]
one-tailed p-value [ 0.20892925]


In [8]:
# conceptual check for one-tail
sp.stats.t(39).sf(fat_t_stat)

array([ 0.20892925])

### 3.1 Commentary

We are testing if we can reject the burger place's claim that the fat content of their burgers is no more than 20%. We run our t-test to find the t-stat of our sample mean at 20.42, and we get 0.81, with two-tailed p-value of 0.42.  Therefore, since our original two-tailed test tells us the combined p-values on both ends of the distribution, we divide by two to find the p-value of a burger having fat content higher than the sample mean.

Since the $\text{p-value} = P(t > 0.82) = 0.21 \Rightarrow \text{ fail to reject } H_0$

In other words, there is roughly a 20% of getting a result more extreme than our sample mean. Therefore, we can't reject the null.

We are assuming a 5% significance level.

In [36]:
#With a month to go in the election, the opinion poll shows that the Democratic candidate is
#leading with 55% support. You want to know if the true percentage is different from 55%.
elect = pd.read_csv('https://s3-us-west-2.amazonaws.com/dsci/6002/data/election.csv')
elect_t_stat, elect_two_tailed_p_value = ttest_1samp(elect, .55)
elect.mean()
np.count_nonzero(elect)
# print("t-stat", elect_t_stat)
# print("mean:", elect.mean()[0])
# print("two-tailed p-value", elect_two_tailed_p_value)
elect_z_statistic, elect_two_tailed_p_value = proportions_ztest(np.count_nonzero(elect), len(elect), .55)
print("z-stat", elect_z_statistic)
print("mean:", elect.mean()[0])
print("two-tailed p-value", elect_two_tailed_p_value)

z-stat -6.74967868609
mean: 0.402
two-tailed p-value 1.48172925761e-11


### 3.2 Commentary

We are testing if Democratic candidate truly has 55% or not. We run our z-test to find the z-stat of our sample mean at 44%, and we get -6.749, with two-tailed p-value of 1.48 x e^-11, an almost zero answer.  

Since the $\text{p-value} = P(t \neq 0.82) = 1.48 x e^-11 \Rightarrow \text{ reject } H_0$

In other words, there is an almost zero chance of getting a result more extreme than our sample mean. Therefore, we can reject the null: The Democratic candidate does not have close to 55% support.

We are assuming a 5% significance level.

In [43]:
#3. Apple Inc. is interested to find out if the battery life of a 
# Macbook Pro that has been used for a month is more than 48 hours.
batt_life = pd.read_csv('https://s3-us-west-2.amazonaws.com/dsci/6002/data/battery_life.csv')
batt_life.mean()
# print("t-stat", elect_t_stat)
# print("mean:", elect.mean()[0])
# print("two-tailed p-value", elect_two_tailed_p_value)
batt_life_t_stat, batt_life_two_tailed_p_value = ttest_1samp(batt_life, 48)
print("t-stat", batt_life_t_stat)
print("mean:", batt_life.mean()[0])
print("two-tailed p-value", batt_life_two_tailed_p_value)

t-stat [ 7.34153004]
mean: 50.1723590003
two-tailed p-value [  5.99666418e-11]


### 3.3 Commentary

We are testing if to see if battery life for a used Macbook Pro is more than 48 hrs. We run our t-test to find the t-stat of our sample mean at 50.17, and we get 7.34, with two-tailed p-value of 5.99 x e^-11.  Therefore, since our original two-tailed test tells us the combined p-values on both ends of the distribution, we divide by two to find the p-value of a burger having fat content higher than the sample mean. 

Since the $\text{p-value} = P(t > 48) = 5.99 x e^-11 \Rightarrow \text{ reject } H_0$

In other words, there is an almost zero chance of getting a result more extreme than our sample mean. Therefore, we can reject the null: The battery life of a used Macbook Pro does not operate more than 48 hrs.

We are assuming a 5% significance level.

**Question 4**

A diet doctor claims that the average American is more than 10 pounds overweight. To test his claim, a random sample of 50 Americans was weighed, and the difference between their actual weight and their ideal weight was calculated. He found that $\bar{x} = 11.5$ and $s = 2.2$ pounds. 

1) Can we conclude that the doctor’s claim is true?  

2) What are the Type I and Type II errors?  

3) Suppose that the average American is about 13 pounds overweight, what is the power of the above test?

### Question 4.1

$H_o: µ \leq 10 \\ H_a: µ > 10$

In [57]:
pounds_t_stat = (11.5 - 10)/(2.2/np.sqrt(50))
sp.stats.t(49).sf(pounds_t_stat)

7.1217429013529294e-06

In [50]:
pounds_t_stat


4.8211825989991874

Yes, we would conclude that the doctor's claim is true because we would reject the null hypothesis. Our p-value is significant. 

### Question 4.2 

Type I: We falsely reject the null and accept the doctor's claim. Average americans are not more than 10 lbs overweight, but we falsely concluded they were.

Type II: We falsely accept the null and reject the alternative. Average americans are, in fact, more than 10 lbs overweight, but we falsely concluded that they were healthy.

### Question 4.3

$$\beta = \text{p-value of 13 lbs t-stat} = 7.1217429013529294e^{-06} $$

$$ \beta = P(\text{Type II error}) = P(\text{fail to reject } H_0 | H_1 \text{ is true}) \\ power = P(\text{reject } H_0 | H_1 \text{ is true}) = 1 - \beta $$



In [53]:
moar_pounds_t_stat = (11.5 - 13)/(2.2/np.sqrt(50))
Beta = sp.stats.t(49).cdf(moar_pounds_t_stat)
Beta

7.1217429013529294e-06

In [54]:
power = 1 - Beta
power

0.99999287825709859

**Question 5**

A company that claims the average time a customer waits on hold is less than 5 minutes. A sample of 35 customers has an average wait time of 4.78 minutes with a standard deviation for wait time is 1.8 minutes. Test the company's claim.

### Question 5

$H_o: µ < 5 \\ H_a: µ \geq 5 \\ n = 35, \ \bar{x} = 4.78, \ s = 1.8$

In [58]:
wait_t_stat = (4.78 - 5)/(1.8/np.sqrt(35))
sp.stats.t(34).cdf(wait_t_stat)

0.23728906420569901

#### Commentary:

Since the $\text{p-value} = P(t < 4.78) = 0.237 \Rightarrow \text{ fail to reject } H_0$

The company's claim that their wait time is less than 5 mins still holds ground. We can't reject the null that wait time is less than 5 mins. 

**Question 6**

The manufacturer of the Bic Extended Lighter claims that it lights on the first time 75% of the time. Test this claim.  
Suppose we make 300 attempts and the lighter lights on the first try 214 times.

### Question 6

$H_o: µ = .75 \\ H_a: µ \neq .75 \\ n = 300, \ \bar{x} = .7133$

$$ z_o = \frac{\hat{p} - p_o} {\sqrt{\frac{{p}_o(1 - {p}_o)}{n}}} $$  

In [68]:
light_n = 300
firsts = 214
light_p_hat = firsts/light_n
light_p_hat

0.7133333333333334

In [69]:
light_z_statistic, light_two_tailed_p_value = proportions_ztest(firsts, light_n, .75)
print("z stat", light_z_statistic)
print("p value", light_two_tailed_p_value)

z stat -1.40442062963
p value 0.160193631336


#### Commentary
the p value is not significant (p > 0.05), so we can't reject the claim that the lighters ignite 75% of the time.

**Question 7**  

A government bureau claims that more than 50% of U.S. tax returns were filed electronically last year.  
A random sample of 150 tax returns for last year contained 86 that were filed electronically. 

Test the government's claim and state the Type I and Type II errors of the test.

### Question 7

$H_o: µ > 5 \\ H_a: µ \leq .50 \\ n = 150, \ \bar{x} = .7133$

In [72]:
ret_n = 150
filed = 86
ret_p_hat = filed/ret_n
ret_p_hat

0.5733333333333334

In [73]:
ret_z_statistic, ret_two_tailed_p_value = proportions_ztest(filed, ret_n, .5)
print("z stat", ret_z_statistic)
print("p value", ret_two_tailed_p_value)

z stat 1.8159299943
p value 0.069381111549


#### Commentary

the p value is insignificant (p > 0.05), so we can't reject the claim that 50% of US tax returns were filed electronically last year.